# ML-02 — Research Question and Provisional Lane

[![Open In Colab](https://colab.research.google.com/assets/colab-badge.svg)](https://colab.research.google.com/github/MohammadAqib-spec/flyrank-ml-internship/blob/main/work/notebooks/w01_research_question.ipynb?flush_cache=true)

This skeleton is yours to fill. Work the sections **in order** — each one has a one-line hint. Simple words, honest numbers.

> Working with an AI assistant? Tell it to read `skills/README.md` first and load the one skill this assignment names on its card.

## 1. My lane (or freestyle) and why

*Name your lane — or say 'freestyle' and describe your own question. One short paragraph: why this one?*

**Provisional lane: Content Refresh / Decline Detection**

I am choosing the content refresh and decline detection lane. The goal is to identify pages whose observed search performance suggests that they may need human review or a content refresh. This lane is useful because the available search data contains signals such as clicks, impressions, CTR, average position, and historical performance. Instead of starting with a model, I want to first understand which signals are associated with pages that appear to be declining and how those signals could support a practical content-review decision.


## 2. The question: decision, action, cost of a wrong call

*What decision does your work improve? Who acts on it? What does a wrong recommendation cost?*

**Research question:**

Can the available search-performance signals help us rank pages by their likelihood of needing a content refresh, so that a content team can decide which pages should be reviewed first?

**Decision:**
Which pages should the content team prioritize for human review?

**Action:**
Review the highest-priority pages and decide whether to refresh, improve, monitor, or leave the content unchanged.

**Cost of a wrong call:**
If a page that needs attention is missed, the team may lose an opportunity to improve its search performance. If a healthy page is incorrectly prioritized, limited content-team time may be spent reviewing a page that does not need a change.


## 3. Quick look at the data (2-3 real numbers)

*Load the starter CSV below and show 2-3 real numbers that make your lane look worth the next 7 weeks.*

In [8]:
import pandas as pd

url = "https://raw.githubusercontent.com/flyrank-bih/flyrank-ml-internship-starter/main/data/raw/content_refresh_anonymized.csv"

df = pd.read_csv(url)

print("Rows:", len(df))
print("Columns:", len(df.columns))
print("\nColumn names:")
print(df.columns.tolist())

Rows: 30000
Columns: 44

Column names:
['content_id', 'client_id', 'search_volume', 'competition', 'competition_level', 'cpc', 'content_type', 'main_intent', 'word_count', 'char_count', 'provider_used', 'model_used', 'impressions_90d', 'clicks_90d', 'pageviews_90d', 'sessions_90d', 'users_90d', 'engaged_sessions_90d', 'ai_sessions_90d', 'scroll_events_90d', 'days_with_impressions', 'days_with_sessions', 'impressions_last_30d', 'clicks_last_30d', 'sessions_last_30d', 'impressions_prev_30d', 'clicks_prev_30d', 'sessions_prev_30d', 'content_age_days', 'age_tier', 'age_tier_order', 'days_since_last_update', 'freshness_tier', 'word_count_tier', 'char_count_tier', 'ctr', 'avg_position', 'engagement_rate', 'scroll_rate', 'ai_traffic_pct', 'impression_tier', 'position_tier', 'trend_direction', 'trend_pct']


In [9]:
print("Total pages:", len(df))

print("\nPages by trend direction:")
print(df["trend_direction"].value_counts())

print("\nAverage CTR:", round(df["ctr"].mean(), 4))
print("Average position:", round(df["avg_position"].mean(), 2))

Total pages: 30000

Pages by trend direction:
trend_direction
down      16262
stable     5962
up         4388
new        2236
flat       1152
Name: count, dtype: int64

Average CTR: 0.5107
Average position: 16.34


The starter dataset contains **30,000 pages** across **44 columns**. In the observed data, **16,262 pages have a “down” trend**, compared with **5,962 stable pages** and **4,388 pages with an “up” trend**. The average CTR is **0.5107**, and the average position is **16.34**.

These observed numbers make the content refresh and decline detection lane worth investigating because a substantial number of pages are marked as declining. The next step is to understand which measurable signals are associated with these observed trends and whether they can help prioritize pages for human review.


## 4. Careful words: what I can and can't claim

*Write what your work will be able to say (observed, directional, decision-support) — and what it never will (causal proof, 'predicting Google').*

### What this work can and cannot claim

This work can identify observed patterns in search-performance data and rank pages that appear more likely to need human review. The analysis can provide directional evidence about which measurable signals are associated with observed declines, such as changes in impressions, clicks, sessions, CTR, and average position.

The output should be treated as decision-support rather than a final decision. A high-priority page is a page that may deserve human review, not a page that is guaranteed to need a content refresh.

This work cannot prove that a particular change in content caused a change in search performance. It also cannot prove that a page will decline in the future or predict exactly how Google will rank a page. Search performance can be affected by many factors that are not available in this dataset.

Therefore, the results will be used to support human review and prioritization, not to make causal claims or claim that the model can predict Google's decisions.


In [10]:
# This cell is for CODE (numbers, a query, a check).

## Self-check

Before you submit, confirm each line honestly:

- [ ] Every section above is filled — markdown thinking AND the code that backs it
- [ ] The notebook runs top to bottom with no errors (Runtime → Run all)
- [ ] No client names, URLs, or private queries anywhere
- [ ] My claims use careful words: observed, measured, directional, decision-support
- [ ] Committed to my repo under `work/notebooks/` — then submit your repo URL on the card. Done.